src/utils.js

```diff
@@ -30,11 +30,26 @@ export function windowOpen(url, { name, height = 400, width = 550 }) {
     chrome: 'yes',
   };
 
-  return window.open(
+  const shareDialog = window.open(
     url,
     platform.name === 'IE' && parseInt(platform.version, 10) < 10 ? '' : name,
     Object.keys(config).map(key => `${key}=${config[key]}`).join(', ')
   );
+
+  const interval = window.setInterval(() => {
```

Comment:
I'd suggest to use [`onclose` callback](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onclose) on sharedDialog.

```js
if (closeCallback && shareDialog !== null) {
    shareDialog.onclose = closeCallback;
}
```

Remember that If the window couldn't be opened, the returned [value is null](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Return_value).